### ISSUE TITLE
should the initial guess be whitened too?

### ISSUE THREAD

**Issue by gaika** [JuliaStats/MultivariateStats.jl#118]
https://github.com/JuliaStats/MultivariateStats.jl/blob/master/src/ica.jl#L169

Should `W = W0'winit` when `do_whiten == true`? Otherwise the sizes do not match.

**Comment by wildart** [comment]
Did you mean this? https://github.com/JuliaStats/MultivariateStats.jl/blob/2c614297917812325411f9a039cc1811d931169c/src/ica.jl#L175-L177

**Comment by gaika** [comment]
no, when the input `winit` is given, what is the size of it? It should be m * k, right? Then `fastica!` will complain that the size of [`W` doesn't match the size of `X`](https://github.com/JuliaStats/MultivariateStats.jl/blob/2c614297917812325411f9a039cc1811d931169c/src/ica.jl#L67). I just noticed that `W0'winit` has the right size and makes the output `W` similar to `winit` when the input is slightly perturbed.

**Comment by wildart** [comment]
It's a bug. If you know how to fix it, PR will be appreciated.
